Sync v2-5-stable with v2-5-test to release 2.5.2

ephraimbuddy opened the pull request:
Time for `2.5.2rc1`!

Taragolis committed:
Migrate remaining core sensors tests to `pytest` (#28204)

(cherry picked from commit e93b669d841828b5f912edf40f255299b1833521)

potiuk committed:
More robust cleanup of executors in test_kubernetes_executor (#28281)

As a follow up after #28047, this PR will make the test cleanup
more robust and resilient to any errors that might have caused
kubernetes_executors left behind.

wrapping start()/end() in try/finally will make the tests
completely resilient to cases where the asserts start to fail -
without those, any failure in tests would cause the same resource
leakage as we initially had when #28407 was iterated on.

(cherry picked from commit 3b203bcb676853bd642a01121988b1cbe929307d)

potiuk committed:
Move provider issue generation to breeze (#28352)

The exercise where another PMC performed a release process had
proven useful to show the value of Breeze in release management
commands - there were no problems with running the Breeze commands,
but for scripts put elsewhere (docs building, dev) it was a bit
problematic because each required own virtualenv.

This is the first PR to simplify that - one less venv for the
issue generation process - now it is moved to Breeze so no more
setup is needed.

(cherry picked from commit fd5846d256b6d269b160deb8df67cd3d914188e0)

Taragolis committed:
Refactor python operators/sensor tests (#28493)

(cherry picked from commit 884fca8d114ce8e0c982747937a1014f3b5e7491)

potiuk committed:
Fix bugs introduced when moving the code of issue generation to breeze (#28683)

The #28352 introduced subtle bugs which failed when runnint it during
provider generation.

Also added running the command automatically in CI.

(cherry picked from commit 69ab7d8252f830d8c1a013d34f8305a16da26bcf)

pierrejeambrun committed:
Update doc for API clients release policy (#28521)

* Update doc for API clients release policy

* Update README.md

Co-authored-by: Xiaodong DENG <xd.deng.r@gmail.com>

* Update airflow release process

* Clarify patch release for clients, add process link

* Update dev/README_RELEASE_AIRFLOW.md

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

Co-authored-by: Xiaodong DENG <xd.deng.r@gmail.com>
Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>
(cherry picked from commit 654242b54638d7a682246d9dbea394bbb59d8923)

dependabot[bot] committed:
Bump swagger-ui-dist from 3.52.0 to 4.1.3 in /airflow/www (#28824)

Bumps [swagger-ui-dist](https://github.com/swagger-api/swagger-ui) from 3.52.0 to 4.1.3.
- [Release notes](https://github.com/swagger-api/swagger-ui/releases)
- [Commits](https://github.com/swagger-api/swagger-ui/compare/v3.52.0...v4.1.3)

---
updated-dependencies:
- dependency-name: swagger-ui-dist
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
(cherry picked from commit e1e3a8e94dd29630a8912aab5c1bf0522c2ec9aa)

dstandish committed:
Throttle streaming log reads (#28818)

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit 46704eed5340ef0e71cf828bb560b3d00aa88691)

jedcunningham committed:
Annotate KubernetesExecutor pods that we don't delete (#28844)

We weren't keeping track of which pods we'd finished with yet, so if you
had `[kubernetes_executor] delete_worker_pods` false, your
KubeExecutor would adopt every single remaining pod when starting up.
Every time.

We now annotate them with `airflow_executor_done` when processing a pods event
from the watcher, so we can ignore the pod when doing adoption.

(cherry picked from commit 72da8bff12e3133045b61936952599a84d3f53a2)

dstandish committed:
Add dep context description for better log message (#28875)

Otherwise, it appears that there is a duplicate log record.

(cherry picked from commit 1ca94ee6ba767ed6851858db24319aa1008562eb)

steveyz-astro committed:
clarify that the total_entries property isn't impact by pagination (#28867)

Co-authored-by: Ash Berlin-Taylor <ash@apache.org>
(cherry picked from commit 952952602161d353effa877fbb091cbfcf56068b)

jedcunningham committed:
KubenetesExecutor sends state even when successful (#28871)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 8a9959cc1ead49785b10b9b28c101e3d94cb4176)

potiuk committed:
Switch pull-request-target to base branch of pull request (#28921)

When pull request target workflow runs, we used build scripts from
main - to couple it with github workflows, but GitHub actually uses
base branch now for the workflows, so we should also use the target
branch for that.

This PR changes it and also extracts the code to do the checkout
to a separate composite action.

(cherry picked from commit 50f20037e8793913c6da2b296462a744dd02e23c)

dstandish committed:
Don't get ES log template from airflow local settings unless necessary (#28882)

This creates an unnecessary warning.

(cherry picked from commit c5ee4b8a3a2266ef98b379ee28ed68ff1b59ac5f)

Taragolis committed:
Allow URI without authority and host blocks in `airflow connections add` (#28922)

(cherry picked from commit d8b84ce0e6d36850cd61b1ce37840c80aaec0116)

kaxil committed:
Fix inconsitencies in checking edit permissions for a DAG (#20346)

We were short-circuting permission in the views instead of letting the security manager handle that. A user will find it inconsistent as the Graph and other views check "per-dag" permissions via https://github.com/apache/airflow/blob/174681911f96f17d41a4f560ca08d5e200944f7f/airflow/www/views.py#L579

so if someone uses Custom Security Manager that will end up with user not being able to "pause" DAG from individual dag page but would be able to do so from Homepage. This PR fixes this inconsistency and gives back this responsibility of permissions to security manager instead if Views.

(cherry picked from commit 87ea8acd0c8e7339249bec51908b35e5e3d70129)

gxcuit committed:
improve quick start guide (#28949)

change task run to task test

(cherry picked from commit a4f6f3d6fe614457ff95ac803fd15e9f0bd38d27)

pierrejeambrun committed:
Update go client gen command (#28967)

(cherry picked from commit 2eb4e974d561db01d0160148b3e372998409d377)

potiuk committed:
Bump epoch on python dependencies in Airlfow Image (#28960)

Recently some packages were removed from Airflow but they are still
available in cached image in Github and on clean install of the
image it causes conflicting dependencies when `pip check` is run.

This change will invalidate cache and reinstall base dependencies
from the scratch - thus removing some dependencies.

(cherry picked from commit fe46a934fc298ec3ac74e6c40ca7e1872d2482f9)

uranusjr committed:
Mark license block in doc as text (#28965)

Sphinx was helpfully (but incorrectly) adding Python syntax highlighting
to the block. This marks the block explicitly as pure text to avoid
that.

(cherry picked from commit 4fff0555b505f7a51efb2bbd108378780eb850d9)

uranusjr committed:
Resolve all variables in pickled XCom iterator (#28982)

(cherry picked from commit ccf53e167ea57716c76ec7ab5bd1223f0c0d47d3)

ephraimbuddy committed:
Remove erroneous TODO comment (#29015)

The issue that this comment was referring to was closed as won't fix.
This PR removes the comment.
Issue: https://issues.apache.org/jira/browse/AIRFLOW-1455

(cherry picked from commit e3a6bd817f3c32ec5e780449090ad96782267d80)

jedcunningham committed:
Be more selective when adopting pods with KubernetesExecutor (#28899)

* Be more selective when adopting pods with KubernetesExecutor

When trying to adopt "resettable" TIs from SchedulerJob, we should not
list out all the pods to compare against, only those that didn't
succeed. This means we will get any pods that are still starting,
running, or failed (meaning the TI wasn't moved to a terminal state
there, and will be in out "adoptable" list).

This avoids the scenario where a dead scheduler has both a completed,
successful worker, and a still running worker, causing log lines
like these about the successful one:

    ERROR - attempting to adopt taskinstance which was not specified by
    database: TaskInstanceKey(...)

This also makes sure we only find pods with the
`kubernetes_executor=True` label for extra safety.

Closes #28071

* Also ignore done pods

(cherry picked from commit f64ac5978fb3dfa9e40a0e5190ef88e9f9615824)

Commit:
change context example to no longer us as variable (#29021)

https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#dags-used-in-a-context-manager-no-longer-need-to-be-assigned-to-a-module-variable-23592
(cherry picked from commit bc5cecc0db27cb8684c238b36ad12c7217d0c3ca)

SoxMax committed:
Update how PythonSensor returns values from python_callable (#28932)

* Update how PythonSensor returns values from python_callable

* test poke returns the xcom value

* update test to only run poke

* reformat based on changes

* use full if rather than ternary

(cherry picked from commit b0f302e027d09a50493f4cdd808559984e433ee1)

potiuk committed:
Sanitize url_for arguments before they are passed (#29039)

The url_for of flask has special arguments that start with `_` and we
should sanitize the ones that come with the request before passing them.

(cherry picked from commit 7f2b065ccd01071cff8f298b944d81f3ff3384b5)

BasPH committed:
Move project and license docs down in menu to start with developer-focused docs (#28956)

(cherry picked from commit d03b9a76914babef23b84acdb6061b0b93bdc2e3)

BasPH committed:
Annotate and simplify code samples in DAGs doc (#29027)

(cherry picked from commit 80dbfbc7ad8f63db8565baefa282bc01146803fe)

potiuk committed:
Fix kerberos authentication for the REST API. (#29054)

Previously we assigned kerberos user name directly to the flask user,
but this had no chance to work because we expect FAB user there and
our security code crash with 'str' has no attribute 'perms'.

This PR uses Kerberos username (including the Kerberos realm) to
retrieve the user from the security manager. This means that
the user name has to have the form of `user_name@KERBEROS_REALM`.

The reason why we are not using email (despite similarities of
the realm and domain name is that those are often different. Email
domain names have often nothing to do the with the realms within
organisations, and it seems safer to put fully qualified names
including the realm in order to uniquely identify the users in
case the organisation uses more than one REALM.

Fixes: #28919

Co-authored-by: BMFH <bogner85@mail.ru>

Co-authored-by: BMFH <bogner85@mail.ru>
(cherry picked from commit 135aef30be3f9b8b36556f3ff5e0d184b0f74f22)

surabathini committed:
logging poke info when external dag is not none and task_id and task_ids are none (#28097)

(cherry picked from commit 760c52949ac41ffa7a2357aa1af0cdca163ddac8)

potiuk committed:
Fix pre-commit warning for exclude in inclusive-language check (#29057)

The exclude regexp contained /* which was not really what we meant
(no harm done but latest pre-commit started to warn about it).

(cherry picked from commit f114c67c03a9b4257cc98bb8a970c6aed8d0c673)

BasPH committed:
Capitalize dag to DAG (#29064)

(cherry picked from commit 10f0f8bc4be521fd8c6cdd057cc02b6ea2c2d5c1)

bharat99k committed:
fixing import error for dataset (#29007)

(cherry picked from commit 481f27170673cb1e4fc8210341c8d341b11925e9)

Bowrna committed:
listener plugin example added (#27905)

(cherry picked from commit 100bb8d79a1e0c5fe6fca4b69c529b447cc992d1)

bbovenzi committed:
Check for run_id url param when linking to graph/gantt views (#29066)

(cherry picked from commit 9dedf81fa18e57755aa7d317f08f0ea8b6c7b287)

hussein-awala committed:
emit dagrun failed duration when timeout (#29076)

(cherry picked from commit ca9a59b3e8c08286c8efd5ca23a509f9178a3cc9)

BasPH committed:
Several improvements to the Params doc (#29062)

* Several improvements to the Params doc

* Rewrite first paragraph for clarity

* Update docs/apache-airflow/core-concepts/params.rst

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit af9143eacdff62738f6064ae7556dd8f4ca8d96d)

potiuk committed:
Only skip provider integration tests for non-main builds (#29022)

Previously we skipped all integration tests in non-main builds (i.e.
builds that were running in v2-* branches where we test and prepare
releases. This had two effects:

* the non-provider integration tests were not run before release, which
  could have revealed some problems
* constraint generation was skipped for those runs because constraint
  generation depended on integration tests

This is now fixed:

* integration tests are run on non-main builds
* but all provider tests are skipped there

(cherry picked from commit c04cfec6de68858d3372f16749244c34021310df)

potiuk committed:
Make options output of breeze command consistent and complete (#29097)

There were a anumber of cases where the "--help" output of Breeze
command had shown a bit inconsistent/incomplete output for rich-click
generated help. The options are assigned to groups to make it easier
to understand what those options do and to make it easier to digest,
but the problem was that if you forgot to add an option to a group, it
was added to "common" options and it was quite misleading.

Also some left-over options were left in the rich-click groups when the
options have been removed. It had no visible effect, but they were
polluting the code with inaccurate information.

This PR fixes all those cases in the past where it happened but also
prevents the cases from happening in the future by adding a new
(yes you guessed it) Breeze command that performs the check and fails
in case any inconsistencies are found (with helpful messsage explaining
the user what to do. This command is run as part of our CI so it will
prevent similar problems from re-appearing.

(cherry picked from commit 9bc48747ddbd609c2bd3baa54a5d0472e9fdcbe4)

potiuk committed:
Make options output of breeze command consistent and complete (#29097)

There were a anumber of cases where the "--help" output of Breeze
command had shown a bit inconsistent/incomplete output for rich-click
generated help. The options are assigned to groups to make it easier
to understand what those options do and to make it easier to digest,
but the problem was that if you forgot to add an option to a group, it
was added to "common" options and it was quite misleading.

Also some left-over options were left in the rich-click groups when the
options have been removed. It had no visible effect, but they were
polluting the code with inaccurate information.

This PR fixes all those cases in the past where it happened but also
prevents the cases from happening in the future by adding a new
(yes you guessed it) Breeze command that performs the check and fails
in case any inconsistencies are found (with helpful messsage explaining
the user what to do. This command is run as part of our CI so it will
prevent similar problems from re-appearing.

(cherry picked from commit 9bc48747ddbd609c2bd3baa54a5d0472e9fdcbe4)

ashb committed:
Fix warning in migrations about old config. (#29092)

We got this warning otherwise:

> .../0069_2_0_0_add_scheduling_decision_to_dagrun_and_.py:66 FutureWarning: section/key [core/dag_concurrency] has been deprecatedyou should use[core/max_active_tasks_per_dag] instead. Please update your `conf.get*` call to use the new name

(cherry picked from commit 9c3cd3803f0c4c83b1f8220525e1ac42dd676549)

auvipy committed:
Remove upper bound limitation for `pytest` (#29086)

(cherry picked from commit c0f6bfa35b82f9a756b31bbba739c9918a28c771)

schwartzpub committed:
Update set-up-database.rst (#29104)

Add experimental notation to MSSQL to be consistent with previous documentation.

(cherry picked from commit c01f9f8c9256e400725da6599e2e617a7cdcd438)

Commit:
PR: Introduced fix to run tasks on Windows systems (#29107)

* Introduced fix to run tasks on Windows

Co-authored-by: Dominik Werner <dwerner@casra.ch>
(cherry picked from commit f8e8fc02fe0598ad7d00c9e95f042371d2ca38f8)

ephraimbuddy committed:
Improving the release process (#27829)

* Improving the release process

This PR aims to automate some parts of the release process.

For RC, we have a command to run the release process:

  breeze release-management start-rc-process --version 2.4.3rc1 --previous-version 2.4.2

For release, the below command runs the process:

  breeze release-management start-release --release-candidate 2.4.3rc1 --previous-release <PREVIOUS RELEASE>

And there's a command for creating a version branch when releasing a minor version:

  breeze release-management create-minor-branch --version-branch ${VERSION_BRANCH}

Use breeze's user_confirm & get_console.print

Skip some steps in CI

Run command in CI

add user_confirm_bool function, a version of user_confirm

Remove retagging of images

simplify console.print calls

Update readme config hash

Use dry-run for commands that shouldn't run in CI

Remove outdated instruction

Remove single quotes on double quote

* fixup! Improving the release process

* fixup! fixup! Improving the release process

* remove hidden arg from commands

* Add check=True to all the run_command to raise when there's a failure

* fixup! Add check=True to all the run_command to raise when there's a failure

* Update minor release command

* Update the ci

(cherry picked from commit 9411e395ecc3bf54ddc341b0f0e227608bd7f972)

snjypl committed:
Fix #28391 manual task trigger from UI fails for k8s executor (#28394)

Manual task trigger from UI fails for k8s executor. the executor.job_id
is currently set to "manual". the task instance queued_by_job_id field
is expected to be None|Integer. this causes the filter query in
clear_not_launched_queued_tasks method in kubernetes_executor to fail
with psycopg2.errors.InvalidTextRepresentation invalid input syntax for integer: "manual" error.

setting the job_id to None fixes the issue.

(cherry picked from commit 9510043546d1ac8ac56b67bafa537e4b940d68a4)

bbovenzi committed:
Clarify graph view dynamic task labels (#29042)

(cherry picked from commit 33ba242d7eb8661bf936a9b99a8cad4a74b29827)

jedcunningham committed:
General cleanup of things around KubernetesExecutor (#29148)

The biggest change here is using `State` or `TaskInstanceState` where
appropriate. Even though their strings are the same, it provides useful
context, so we should use the correct enum.

One could argue `State` isn't the right thing to use, but at least it's
consistently wrong now :)

(cherry picked from commit bfbe2cb6fc39ccc24a13befa28e63beccbbf37b0)

maxnathaniel committed:
Update broken link for VS Code Quick Start guide (#29154)

(cherry picked from commit b314db9880ca9936cbe82e5527bf80d34d3d7861)

ncorriveau committed:
fixed et.c type to etc. (#29163)

(cherry picked from commit c18dbe963ad87c03d49e95dfe189b765cc18fbec)

eladkal committed:
Add release flag to Breeze generate-issue-content (#28952)

This is useful to use one command to generate issues for all our releases.

(cherry picked from commit cebeb68d8b5c2e8867fe3ac9f2a88dd9b5662568)

jedcunningham committed:
Use `pod_name` when patching done KE pods (#29159)

This isn't in a release yet, so let's take the opportunity to fix the
naming of the kwarg.

(cherry picked from commit b8f15a98bf943884262a498df66b81268fa71a98)

dependabot[bot] committed:
Bump ua-parser-js from 0.7.31 to 0.7.33 in /airflow/www (#29172)

Bumps [ua-parser-js](https://github.com/faisalman/ua-parser-js) from 0.7.31 to 0.7.33.
- [Release notes](https://github.com/faisalman/ua-parser-js/releases)
- [Changelog](https://github.com/faisalman/ua-parser-js/blob/master/changelog.md)
- [Commits](https://github.com/faisalman/ua-parser-js/compare/0.7.31...0.7.33)

---
updated-dependencies:
- dependency-name: ua-parser-js
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
(cherry picked from commit 2e90c06c3eaf6a8cd0d7f4055274834158879440)

Taragolis committed:
Fix integration with `codecov` (#29145)

(cherry picked from commit 30ad26e705f50442f05dd579990372196323fc86)

ephraimbuddy committed:
Remove misleading docstring on dagrun_timeout (#29169)

This docstring is no longer correct (since 2.0) and should be removed

(cherry picked from commit a72e28d6e1bc6ae3185b8b3971ac9de5724006e6)

hussein-awala committed:
redirect to the origin page with all the params (#29212)

(cherry picked from commit 7315d6f38caa58e6b19054f3e8a20ed02df16a29)

pierrejeambrun committed:
Update package version for go and python client (#29218)

(cherry picked from commit 38d0f36fb4eadb0da50c47d3c1e778b9fc25cea2)

potiuk committed:
Make skipping provider tests part of selective-checks (#29216)

The "SKIP_PROVIDER_TESTS" was added as quick-fix when releasing
2.5.0 release, now we can make it a bit more consistent with
all other selective checks. It has now unit tests covering it
and it is produced as selective-checks output so that we do not
have to encode the logic in conditions in the ci.yml.

(cherry picked from commit e5d51fc50f6aad0f35f775367bc700708fe6d62b)

pierrejeambrun committed:
Explicit a few steps in the release process (#29098)

* Update the release process.

* Fix conflics, small update

* Update go release process link with the newly added resource

(cherry picked from commit bc80313a98e304196795c310db6d842f14031c61)

amarlearning committed:
Add trigger process to Airflow Docker docs (#29203)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 272f358fd6327468fcb04049ef675a5cf939b93e)

potiuk committed:
Fix dev-mode start-airflow in Breeze (#29225)

The recent asset compilation fix #29185 broke --dev-mode flag in
Breeze. The start-airflow command checks if the output file
is successfully deleted, but in case of dev-mode - it is never
deleted because asset compilation runs in background continuously.

The fix is to use different output file for --dev-mode asset
compilation and to make sure it will also delete the regular output
(potentially from the previous run).

(cherry picked from commit 55035878e587631772f078cae0206cbcbef5cd37)

potiuk committed:
Clarify description of worker control config (#29247)

The description mentioned SQS which is not currently officially
supported as broker. However this config could be used in case
of (potentially) any broker that does not support remote control.

In order to not make false impression that SQS is supported, we
should remove the specific reference.

(cherry picked from commit 5a1dc9e6f8f573f1662e10e026bf5bac0b9dd265)

potiuk committed:
Fix commit title for constraint commits (#29236)

The ##29215 nicely fixed the content of commit message but there
is one more fix needed to fix the commit message.

(cherry picked from commit 0369e96d1781b7aee6b2ad2bde39b95089f49f72)

spencer-shadley committed:
Fixed grammatical issue (#29253)

(cherry picked from commit 08ec820b167ab3b2489ad8e4fb900d9390cfea49)

AaronCoquet-Easypark committed:
Change awkward wording about extenibility (#29261)

Alternate wording:
```
The extensibility is one of the many reasons that Apache Airflow is powerful.
```

In either version, the word "so" can be added as an intensifier before "powerful".

(cherry picked from commit 1e046f36f8172f6bf9a4b26e2d3a77aad5eb8bbb)

josh-fell committed:
Use `schedule` param rather than `timetable` in Timetables docs (#29255)

As of 2.4.0, the `timetable` DAG param has been deprecated with preference for `schedule`. The docs for Timetables still reference `timetable` in the code examples.

P.S. Small "bonus" added of removing the `as dag:` registration of DAGs initialized with a content manager too.

(cherry picked from commit 51d96334b8a688851d3ed44f57519fc795a7a509)

vedantlodha committed:
Fix dag run trigger with a note. (#29228)

* Fix dag run trigger with a note.

Currently, triggering dag run with note gives 400. This PR fixes it.

Closes: #28825

* make flask_login.current_user a global import

* precommit error fixes.

* add test coverage for the fix

(cherry picked from commit b94f36bf563f5c8372086cec63b74eadef638ef8)

o-nikolas committed:
Initialize and validate conf early and intentionally (#29257)

Background:

The first time airflow.configuration is imported triggers all the code that creates and validates the conf object to run (since that code lives at the module level in configuration.py). At the moment that first import of airflow.configuration just happens to be in airflow.settings (when settings is imported in airflow.__init__). This has a critical side effect that no code in conf can use or import any code that uses airflow.settings because this would cause an import cycle (since airflow.settings is what imported ariflow.configuration to then trigger the conf object construction). But a TONNE of things in airflow use airflow.settings, so it's very difficult to add or change code in conf without needing to import something that uses airflow.settings.

What this change does:

This tactical code change intentionally imports airflow.configuration in airflow.__init__ before airflow.setttings imports it. This allows the construction and validation of the conf object to happen outside the scope of the airflow.settings module which means it's much easier to add/modify code in conf without causing import cycles (which I need to do for another code change, which started this whole investigation).

(cherry picked from commit 338a633fc9faab54e72c408e8a47eeadb3ad55f5)

kolfild26 committed:
Write action log to DB when DAG run is trigged via API (#28998)

(cherry picked from commit edc2e0b118a77c143b1a5d1eb82f1137148af633)

Adaverse committed:
Fixed link to airflow variable in docs (#29338)

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit d37ef064a4d83b5c9d0d82240d841c62e3b1981f)

bbovenzi committed:
Increase max height for grid view elements (#29367)

(cherry picked from commit 643d736ebb32c488005b3832c2c3f226a77900b2)

bbovenzi committed:
remove /legacy_tree endpoint (#29359)

(cherry picked from commit 307377a98b1106ba3708e9875710834dbf3f6a6a)

potiuk committed:
Remove stderr redirection for get-workflow-info (#29381)

The #28514 change added capability of getting trace errors for
the command by swapping redirection of stderr and stdout, but
debug output for github_output file remoined in stderr.

This was no problem in most cases as rich did not produce color
codes for stderr redirection and github_output was able to parse
it properly - and each line in the GITHHUB_OUTPUT was simply
duplicated.

However, when number of labels was big, rich split the message and
it caused Invalid format message.

(cherry picked from commit 9f1f026fac27a68c48c70a6e5b257b39f9396632)

potiuk committed:
Mark Airflow directory in container as safe for git commands (#29386)

There is a new setting/version of git in GitHub Actions that started
checking the ownership of the Git repository. Since in case of the
provider commands we run them inside docker image as root user
(this is in order to isolate the provider package building from
the main CI environment), the owner of such directory is different
(runner user) than the user that runs the git command (root).

This change marks the current git directory for such commands as
safe, regardles from the discrepancy.

This config is global and run inside the image, so it is safe to
leave it after methods complete as containers are torn-down after
completing package preparation.

This PR also improves diagnostics. Previously the `git remote add`
output was redirected to dev null as there was no way it could fail,
but this turned to be false - the output of the `git remote add`
commnd is now also printed for diagnostics.

(cherry picked from commit 2e1635a9c1d6faaa9d5cd5cfd1a364091cd62bde)

bbovenzi committed:
Fix grid logs for large logs (#29390)

* slice extra long file lines, remove full content

* add warning if logs are truncated

* Add warning for `.split()` errors

* remove extraneous useTaskLog change

* remove Show Full Content checkbox

* remove extraneous console.error

(cherry picked from commit 1ed08028efec9284c490f98df5ce9ba355b673ff)

potiuk committed:
Fix validation of date-time field in API and Parameter schemas (#29395)

The open-api-schema-validator 0.4.3 made RFC3339 validation of
the date-time fields mandatory and this revealed problems in our
test URLs - the '+' was not url-encoded and it was replaced with
space - thus the dates passed were not valid RFC3339 date-time
specifications.

This however revealed one more problem. The RFC3339-validator package
is automatically installed by the open-api-schema-validator, but when
installed, it also adds validation to date-time fields validated by
the Params of ours - for example naive date-time parameters. We
are limiting the supported parameters now to be iso8601-compliant
for backwards compatibility but we deprecate iso8601 in favor of
RFC3339.

This might introduce breaking changes for users who use non-valid
date-time parameters in the API, however we should consider that as a
bugfix, and accidental support, because the date-time schema should
expect RFC3999-formatted date time.

Co-authored-by: Andrey Anshin <Andrey.Anshin@taragol.is>
(cherry picked from commit a1faa2c284828e1c2d4962a37d1173c870f28aac)

potiuk committed:
Remove ssh.rst from private-key check (#29412)

The ssh.rst contains a private key example (legitimately), so the
file should be excluded from private key check.

(cherry picked from commit 9602f1db4b8a3f6b304e631848ccc9fb16790ebb)

josh-fell committed:
Remove `notes` param from TriggerDagRunOperator docstring (#29298)

Related: #27849

As part of #27849, the `notes` functionality was removed from TriggerDagRunOperator. However, the docstring still references this param which might confuse users.

(cherry picked from commit 9bfe02b76a6b3712a4fa5b84e8ba7f73488812e3)

Taragolis committed:
Use non-deprecated method when unpause DAG in the UI (#29475)

(cherry picked from commit c1142a8f417b3250b6ecf598097d841f40ef6a32)

arjunanan6 committed:
Use newer setuptools v67.2.0 (#29465)

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>
(cherry picked from commit 41dff9875bce4800495c9132b10a6c8bff900a7c)

potiuk committed:
Bring back fixed tenacity (#29440)

* Bring back fixed tenacity.

Tenacity MyPy integration has been broken in 8.2.0, but 8.2.1
seems to fix it.

Test fix provided in https://github.com/jd/tenacity/pull/390

Related: #29411

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 28806a32005583f06c8ee26b808d84d2a5593ab7)

potiuk committed:
Require explicit flag to remove MyPy cache when running breeze stop (#29493)

The change #29184 added cleanup of MyPy cache when `breeze stop`
was called, but this is a bit too agressive. It will remove
the cache always when breeze stop is run and it will prolong
(on MacOS by 10s of seconds) the MyPy static checks when it happens.

Since cleaning the cache is only really needed when MyPy cache
gets broken (for example when we upgrade MyPy), cleaning it
explicitly when you encounter such problem is a better idea.

This PR:

* adds flag to breeze stop to clean the mypy cache (disabled by
  default)
* adds documentation about that feature in STATIC_CHECKS
* adds hint displayed to the user when mypy check fails

(cherry picked from commit 8841537e58b864bbfb162692b12cf84f7df29001)

hussein-awala committed:
Fix nested fields rendering in mapped operators (#29451)

(cherry picked from commit 5eefd47771a19dca838c8cce40a4bc5c555e5371)

potiuk committed:
Do not show version/node in UI traceback for unauthenticated user (#29501)

The traceback contains information that might be useful for a potential
attacker to better target their attack (Python/Airflow version, node
name). This information should not be shown if traceback is shown to
unauthenticated user.

(cherry picked from commit cf814550275bd04326f095cc28f93663daf3404b)

michaelmicheal committed:
Ensure Serialized DAG is deleted (#29407)

* dag processing manager, dag serialization delete, only filter on dag folder when dag processing is in a standalone processor

* Remove redundant change

* remove deleted dags from serialized dags, fix None check

* serialized_dag tests, add processor_subdir to remove_deleted_dags test

(cherry picked from commit 02a2efeae409bddcfedafe273fffc353595815cc)

potiuk committed:
Fix circular imports when airflow starts (#29494)

* Fix circular imports when airflow starts from scratch

When airflow CLI is started without generated config file the CLI
fails with circular imports. Some of the util classes imported
during such start were importing "constants" (in fact variables)
from the setttings and this happened before settings were fully
initialized.

This PR moves all relevant top-level imports to be local imports
and moves initialization of the State class with settings-defined
colors to initialization() method in order to avoid those circular
imports.

* Update airflow/models/xcom.py

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 47b67f13da667de4eb69233404d4d5a33574dd58)

akrava committed:
Make `prev_logical_date` variable offset-aware (#29454)

* Make `prev_logical_date` variable offset-aware

* Test Calendar page for DAG with non-cron like schedule

* Use pendulum DateTime.min which is offset-aware

(cherry picked from commit f837c0105c85d777ea18c88a9578eeeeac5f57db)

potiuk committed:
Revert "Google have re-worked their docs site ane appear not publish objects.inv (#28834)" (#29543)

This reverts commit 44d93bea3adbe9429b6fdee14de31f80ac51c3bb.

Google re-published the .inv files after deleting them in January.

Details in https://github.com/googleapis/google-cloud-python/issues/10814

(cherry picked from commit c2c27c34fc76db7bface94affacc82b0d1a34d92)

o-nikolas committed:
Fix import cycle starting from airflow/__main__.py (#29523)

(cherry picked from commit 81f07274b9cd9369a1024eb8b0ad5ee6058202f0)

Jianxuan-Li committed:
detect is 'docker-compose' existing (#29544)

issue #29542

(cherry picked from commit 84b13e067f7b0c71086a42957bb5cf1d6dc86d1d)

potiuk committed:
Remove section check from get-value command (#29541)

* Remove section check from get-value command

The section check is not really needed and harmful. There should
be no problems if sections are missing from the configuration file.

Running this check might lead to a problem that if someone stripds down
the config file to bare minimum and moves the crucial configuration to
env variables the 'get-value' command might fail, even if the
configuration is perfectly sound for Airflow.

Fixes: #29537

* Update airflow/cli/commands/config_command.py

(cherry picked from commit 06d45f0f2c8a71c211e22cf3792cc873f770e692)

jedcunningham committed:
Change expose_hostname default to false (#29547)

(cherry picked from commit f0d90497da83a7518a6c6fc365ea992e9e699bfd)

Taragolis committed:
Improve health checks in example docker-compose and clarify usage (#29408)

* Improve health checks in example docker-compose and clarify usage
* Apply suggestions from code review

---------

Co-authored-by: Bas Harenslak <BasPH@users.noreply.github.com>
(cherry picked from commit 08bec93e5cb651c5180af13bad1c43d763c7583d)

josh-fell committed:
Don't display empty rendered attrs in Task Instance Details page (#29545)

Closes: #29515

The Task Instance Details page was displaying attributes which did not have a value. Mainly these were the docs attributes (i.e. `doc`, `doc_json`, `doc_md`, `doc_rst`, and `doc_yaml`). This clutters the page with less useful information for the task instance.

This PR changes the attributes displayed on the page only if they have non-empty values generally.

(cherry picked from commit f2f6099c5a2f3613dce0cc434a95a9479d748cf5)

dependabot[bot] committed:
Bump undici from 5.9.1 to 5.19.1 (#29583)

Bumps [undici](https://github.com/nodejs/undici) from 5.9.1 to 5.19.1.
- [Release notes](https://github.com/nodejs/undici/releases)
- [Commits](https://github.com/nodejs/undici/compare/v5.9.1...v5.19.1)

---
updated-dependencies:
- dependency-name: undici
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
(cherry picked from commit 792416d4ad495f1e5562e6170f73f4d8f1fa2eff)

vchiapaikeo committed:
Edgemodifier refactoring w/ labels in TaskGroup edge case (#29410)

* Edgemodifier refactoring w/ labels in TaskGroup edge case

(cherry picked from commit 4b05468129361946688909943fe332f383302069)

okayhooni committed:
Handling error on cluster policy itself (#29056)

* handling error except PolicyViolationError on cluster policy

* reflect feedback on catch expr on dagbag

* remove unused imports to satisfy static checks

* fix unittest on invalid timetable exeception

---------

Co-authored-by: ok9897 <ok9897@krafton.com>
(cherry picked from commit 83cdd5031b9683e484be667edec81b5e772ef04e)

potiuk committed:
Specific use-case: adding packages via requirements.txt in compose (#29598)

The users of docker compose do not realise that docker compose
has built-in mechanism for rebuilding the image automatically when
needed and do not realise that this does not make their workflows
more complex. Apparently just describing how you can use custom
images is either scary or not clear enough, as users keep on for
dynamic way of adding the packages (which makes no sense if you
use what docker-compose provides out-of-the-box) so it is
worthwile to specifically mention "requirements.txt" and
explicitly add all the steps needed to make requirements.txt and
custom Docker image part of your regular workflow.

(cherry picked from commit a21c17bc07c1eeb733eca889a02396fab401b215)

hussein-awala committed:
convert moment with timezone to UTC instead of raising an exception (#29606)

* convert moments with timezone to utc instead of raising an exception

* test DateTimeTrigger with different timezones

(cherry picked from commit 79c07e3fc5d580aea271ff3f0887291ae9e4473f)

Thomas-McKanna committed:
Remove poke method to fall back to parent implementation (#29146)

Co-authored-by: Thomas McKanna <mail.thomasmckanna@gmail.com>
(cherry picked from commit 2d3cc504db8cde6188c1503675a698c74404cf58)

rubenbriones committed:
Missing timezone in next_dagrun in parameterized docs (#29362)

(cherry picked from commit e9e3f2c8c75a33504f210529200cf7310ca458ca)

michaelmicheal committed:
datasets, next_run_datasets, remove unnecessary timestamp filter (#29441)

* datasets, next_run_datasets, remove unnecessary timestamp filter

* remove redundant `_and`

(cherry picked from commit 6f9efbd0537944102cd4a1cfef06e11fe0a3d03d)

vincbeck committed:
Fix Rest API update user output (#29409)

* Fix patch user API

* Use UserCollectionItem

(cherry picked from commit 20206e9a934c623b4936a66c1c784083f0f16f71)

curlup committed:
Email Config docs more explicit env var examples (#28845)

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>
Co-authored-by: Bas Harenslak <BasPH@users.noreply.github.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit d306067369d432dc81129947e730fc13de035f76)

josh-fell committed:
Persist DAG and task doc values in TaskFlow API if explicitly set (#29399)

If users set `doc_md` arg on `@dag`- or `@task`-decorated TaskFlow functions and those functions have a docstring, the `doc_md` value is not respected. Instead this PR will enforce only using the TaskFlow function docstring as the documentation if `doc_md` (or any `doc*` attrs for tasks) is not set.

(cherry picked from commit e02bfc870396387ef2052ab375cdd2a54e704ae2)

ephraimbuddy committed:
Fix Scheduler crash when clear a previous run of a normal task that is now a mapped task (#29645)

The fix was to clear the db references of the taskinstances in XCom, RenderedTaskInstanceFields
and TaskFail. That way, we are able to run the mapped tasks

(cherry picked from commit a770edfac493f3972c10a43e45bcd0e7cfaea65f)

aliotta committed:
fix clear dag run openapi spec responses by adding additional return type (#29600)

(cherry picked from commit 54c4d590caa9399f9fb331811531e0ea8d56aa41)

farhan0syakir committed:
introduce dag processor job (#28799)

DagFileProcessorManager was not creating jobs in the metadata DB, so the livenessProbe was not valid.
A new is created for the Standalone DAG Processor.
By doing that, the airflow jobs check --hostname command would work correctly and the livenessProbe wouldn't fail

(cherry picked from commit 0018b94a4a5f846fc87457e9393ca953ba0b5ec6)

josh-fell committed:
FIx formatting of Dataset inlet/outlet note in TaskFlow concepts (#29678)

(cherry picked from commit 9de301da2a44385f57be5407e80e16ee376f3d39)

bbovenzi committed:
Move extra links position in grid view (#29703)

(cherry picked from commit 8449180f4818ecaa5c065c65152922335ffe8c5a)

ecerulm committed:
Fix leak sensitive field via V1EnvVar on exception (#29016)

Co-authored-by: Andrey Anshin <Andrey.Anshin@taragol.is>
(cherry picked from commit 78115c5ca995be6f0e06f68b4d97f7c270c16685)

vemikhaylov committed:
Replace blockquote + Note: with RST note in docs (#29709)

(cherry picked from commit feb4f40db0170e0711234076adf45753d65aed2f)

josh-fell committed:
Update docs re: template_fields typing and subclasses (#29725)

(cherry picked from commit 28573c94ff9054aee76aee42dd1eec314e849016)

potiuk committed:
Change permissions of config/password files created by airflow (#29495)

The permissions for files created by airflow when creating config and
standalone files are now only limited to the owner.

(cherry picked from commit b2ecaf9d2c6ccb94ae97728a2d54d31bd351f11e)

ephraimbuddy committed:
Fix on_failure_callback when task receives a SIGTERM (#29743)

This fixes on_failure_callback when task receives a SIGTERM by
raising a different exception in the handler and catching the
exception during task execution so we can directly run the failure
callback.

(cherry picked from commit 671b88eb3423e86bb331eaf7829659080cbd184e)

EricPickup committed:
Correct version label for data interval variables in documentation (#29704)

* Correct date for data interval variables

* Update prev interval variable docs

(cherry picked from commit 717426ec40bbdd31c564c1d4bdc6ba6e0789d1a2)

hasiotis committed:
Description of dag_processing.last_duration (#29740)

Time metric reports dag_processing.last_duration.<dag_file>  reposts seconds and not milliseconds acording to https://github.com/apache/airflow/blob/2.4.3/airflow/dag_processing/manager.py#L874

(cherry picked from commit 4e3b5ae7248c2327864f64b25dc7a5bd7705430c)

duke8585 committed:
Fixing Task Duration view in case of manual DAG runs only (#22015) (#29195)

* Fixing Task Duration view in case of manual DAG runs only (#22015)

Currently, if a DAG history is solely comprised of manual runs, they cannot be filtered down by `nums` in the duration view, because in this case, the min_date defaults to the current utc_epoch. Consequently, an unlimited number of DAG runs is displayed.
With the implemented fallback, the behavior for mixed (scheduled + manual) DAG runs is similar to the default behavior, as it is for manual-only runs. It was implemented this way to avoid breaking changes.

* Fixing get_task_instances_before method and adding a unit test

With this new fix, the determination of the min_date is done on a set of
execution_date values - via querying - limited by `num`and taking the
last element of it. the fallback value if none are found is simply the
base_date. this way, the invoked get_task_instances cannot default to
30 days prior to the current time.

Furthermore, the query itself is no longer confined to non-manual runs,
but includes all DAGRun types.

More context:

During writing the tests, it became evident
that the current implementation of the get_task_instances_before
method has two problems:
1) With the implementation of a limit + offset determination of the
min_date, no results can occur:
  * Either due to less DAGRuns in the DB than `num` or
  * Because of the constraining to non-manual runs (see more below).
Consequently, the fallback in the invoked get_task_instances is then a
-30d from datetime.now(), so an entirely different time dimension than
the one selected in the UI (execution_date).
2) With the explicit omission of manual runs in the query, in case of n
manual runs and 1 scheduled one before the base_date (e.g.
S-M-M-...-M-M-BD), the scheduled one + n manual runs are returned. in
the case of mainly triggered/manual DAGRuns, this leads to more results
displayed than expected. in case of solely manual DAGRuns, the method
will return no results and hence default to the above mentioned
fallback, which operates on a different timeline. prior tp this fix,
there was even a fallback to the start of the current epochal period,
i.e. showing all DAGRuns in the DB.

With the previously suggested non-breaking fix, it was not possible to
avoid this scenario, with any scheduled run found, the fallback was
never reached.

The suggested changes may be considered
breaking. the changes are confined to the scope of the
get_task_instances_before method and usage is only in the views.py for
the duration, tries and landing times, so where the changes  meant to
fix the existing behavior.

* Update airflow/models/dag.py

Co-authored-by: Andrey Anshin <Andrey.Anshin@taragol.is>

* Update airflow/models/dag.py

Co-authored-by: Andrey Anshin <Andrey.Anshin@taragol.is>

* Update airflow/models/dag.py

Co-authored-by: Andrey Anshin <Andrey.Anshin@taragol.is>

---------

Co-authored-by: Andrey Anshin <Andrey.Anshin@taragol.is>
(cherry picked from commit 8b8552f5c4111fe0732067d7af06aa5285498a79)

pierrejeambrun committed:
Fix compile www assets dev (#29769)

* Fix compile www assets dev

* Add parents True

(cherry picked from commit a0e13370053452e992d45e7956ff33290563b3a0)

Bowrna committed:
better description for limit in api (#29773)

(cherry picked from commit 228d79c1b3e11ecfbff5a27c900f9d49a84ad365)

Josersanvil committed:
Removed continue for not in (#29791)

(cherry picked from commit 60d98a1bc2d54787fcaad5edac36ecfa484fb42b)

john-jac committed:
Add Landing Times entry to UI docs (#29511)

* Add Landing Times entry to UI docs

Explains how Landing Times are calculated and how they relate to task adoption time.  Includes a screenshot of the Landing Times tab.

* Updated to move celery-specific entries to the configuration page

(cherry picked from commit 550c7a44d6b0ad87902caa2fce59cadaa9c2dc28)

amoghrajesh committed:
DAG list sorting lost when switching page (#29756)

* Adding sorting keys to generate_pages

* Handling review comments and using the new keys

* Fixing Brent review comments

* Adding docstring

---------

Co-authored-by: Amogh <adesai@cloudera.com>
(cherry picked from commit c8cd90fa92c1597300dbbad4366c2bef49ef6390)

eladkal committed:
replace tree view with grid view in docs (#29879)

(cherry picked from commit a71f908a0eeab0c0671c39bef394ee07e06039d9)

mayeulk committed:
Correct link to best_practice(s) section (#29876)

Logical link <best_practices/airflow_variables> fails.
In similar file faq.rst, logical link `DAG writing best practices<best_practice:writing_a_dag>` (without "s" at practice) works.

(cherry picked from commit 43d48e17dc18d81d620c9e90758078f152762d74)

josh-fell committed:
Update ref anchor for env var link in Connection how-to doc (#29816)

In the "Managing Connections" how-to doc, the "environment variables" links in the doc point to the "Security of connections in the database" section rather than the "Storing connections in environment variables" section.

(cherry picked from commit b33b11bf9fdbbedbd31699428853896f4db478cb)

potiuk committed:
Move help message to the google auth code (#29888)

The "google-auth" 2.16.2 just released, removed the _HELP_MESSAGE
that was imported by google provider auth util thus failing the
imports in canary builds of ours attempting to upgrade to
newver versions of released libraries.

This PR inlines the original help message into our code to make
it independent from google-auth package version used.

(cherry picked from commit 47ab0ca6295f301d393dbb29e134fa90be0ad7df)

SamWheating committed:
POST /dagRuns API should 404 if dag not active (#29860)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 751a995df55419068f11ebabe483dba3302916ed)

amoghrajesh committed:
Normalising the usage of generate_pages (#29898)

Co-authored-by: Amogh <adesai@cloudera.com>
(cherry picked from commit f0bd45389c7799884a26d2b9c79b0498683d3e03)

chranto committed:
Correct argument name of Workday timetable in timetable.rst (#29896)

(cherry picked from commit c02c3010d83b3cc1cc3007ad56f51af6538c8424)

potiuk committed:
Update min version of python-deamon to fix containerd file limits (#29916)

Recent change in the new containerd causes memory exhaution as
huge amount of memory were used by python-daemon when starting,
thus running Airflow in Docker for multiple OS-es using the new
containerd was impossible without implementing some workarounds.

Python daemon fix has been released in 3.0.0 version in response to
https://pagure.io/python-daemon/issue/72 and we should add min
version for the package to make sure the new version is used.

Fixes: #29841
(cherry picked from commit c8cc49af2d011f048ebea8a6559ddd5fca00f378)

hussein-awala committed:
Add a check for not templateable fields (#29821)

(cherry picked from commit 7963360b8d43a15791a6b7d4335f482fce1d82d2)

potiuk committed:
Limit importlib-metadata backwport to < 5.0.0 (#29924)

The Importlib-metadata 5 breaks importing Celery < 5.3.0. Until
Celery 5.3.0 gets released (it is pre-release/beta now) we should
workaround the problem by limiting importlib-metadata to < 5.0.0.

(cherry picked from commit 4ea457a5ca694eab150e02a26c0b3ef62858fbca)

potiuk committed:
Use python location in cache key used by Breeze (#29947)

Pipx installation is cached between runs and it should use
python location in order to not reuse cache between different
python versions, otherwise pipx installation might fail.

(cherry picked from commit 913b7c113677fc9e444a95ef2071a0ab47312d27)

pierrejeambrun committed:
Update Airflow version to 2.5.2

pierrejeambrun committed:
Update docker image python version to 3.7 in entrypoint doc (#29964)

(cherry picked from commit b4ddb9c79f8d43d981951b4b964663b600d99565)

hussein-awala committed:
Replace deprecated k8s registry references (#29938)

(cherry picked from commit e0fadf9ce981326569cf64f2fbedc9de4a60ee06)

tsoud committed:
Update pipeline.rst - Fix query in `merge_data()` task (#29158)

* Update pipeline.rst - Fix query in `merge_data()` task

The alias for the subquery was missing in the `merge_data()` function's query, so it just needed to be added. Otherwise, the `employees` table remains empty. I found the error in the postgres container logs:
```
2023-01-25 08:00:03 2023-01-25 14:00:03.256 UTC [70428] ERROR:  subquery in FROM must have an alias at character 74
2023-01-25 08:00:03 2023-01-25 14:00:03.256 UTC [70428] HINT:  For example, FROM (SELECT ...) [AS] foo.
2023-01-25 08:00:03 2023-01-25 14:00:03.256 UTC [70428] STATEMENT:
2023-01-25 08:00:03                 INSERT INTO employees
2023-01-25 08:00:03                 SELECT *
2023-01-25 08:00:03                 FROM (
2023-01-25 08:00:03                     SELECT DISTINCT *
2023-01-25 08:00:03                     FROM employees_temp
2023-01-25 08:00:03                 )
2023-01-25 08:00:03                 ON CONFLICT ("Serial Number") DO UPDATE
2023-01-25 08:00:03                 SET "Serial Number" = excluded."Serial Number";
```

Thank you!

* Update docs/apache-airflow/tutorial/pipeline.rst

Co-authored-by: Josh Fell <48934154+josh-fell@users.noreply.github.com>

---------

Co-authored-by: Josh Fell <48934154+josh-fell@users.noreply.github.com>
(cherry picked from commit 0d3d0e2e746dd07ab04752800f0cb7f860f6ac46)

ashb reviewed (dismissed):
182 commits? What on earth is in here?

@pierrejeambrun @ephraimbuddy  we need to be much more picky about what makes it in to point releases.  BUG FIXES ONLY please

This is for three reasons: 

A) semver 
B) but keeping it to bug fixes we keep the release small -> learn like chance of breaking something with a bad cherry pick
C) we don't end up going down the path we had with 1.10 where we were cherry picking too make releases for two years

pierrejeambrun committed:
Add Changelog for 2.5.2

pierrejeambrun committed:
Fix CI

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,113 @@
+
+  In case of parameters, we still allow IS8601-compliant date-time (so for example it is possible that
+  ' ' was used instead of ``T`` separating date from time and no timezone was specified) but we raise
+  deprecation warning. (#29395)
```
```suggestion
In case of API calls, it was possible that "+" passed as part of the date-time fields were not URL-encoded, and
such date-time fields could pass validation. Such date-time parameters should now be URL-encoded (as ``%2B``).

In case of parameters, we still allow IS8601-compliant date-time (so for example it is possible that
' ' was used instead of ``T`` separating date from time and no timezone was specified) but we raise
deprecation warning.
```

Don't need these indented (not sure why recent releases are, PR incoming), nor the PR number duplicated.

pierrejeambrun replied:
Indeed.

The output of:
```
towncrier build --version=${VERSION_WITHOUT_RC} --date=2023-03-14 --dir . --config newsfragments/config.toml
```
Is not `rst` formatted. Indent are list in fact that are removed. So basically we end up editing and formatting this manually, to be consistent with previous releases.
This leave space for inconsistencies. (PR numbers in title, underline, indents etc.)

Same for the body of the changelog which comes from the following command, there is quite a few manual formatting operations here as well.
```
./dev/airflow-github changelog  v2-5-stable  v2-5-test
```

This is something I wanted to ask @ephraimbuddy, maybe there is a better way for building the release note I am not aware of.

pierrejeambrun replied:
Fixed

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,113 @@
+^^^^^^^^^^^^^^^^^^^
+
+The date-time fields passed as API parameters or Params should be RFC3339-compliant (#29395)
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```
```suggestion
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,113 @@
+Default for ``[webserver] expose_hostname`` changed to ``False`` (#29547)
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+  The default for ``[webserver] expose_hostname`` has been set to ``False``, instead of ``True``. This means administrators must opt-in to expose webserver hostnames to end users. (#29547)
```
```suggestion
The default for ``[webserver] expose_hostname`` has been set to ``False``, instead of ``True``. This means administrators must opt-in to expose webserver hostnames to end users.
```

ephraimbuddy committed:
Add release commands to release management command config (#29995)

(cherry picked from commit 24e40a69a70b5da2899081eb0fa9d23a224b63ae)

potiuk committed:
Return error when group is not defined in breeze configuration (#29997)

When group was not found in Breeze configuration, we printed error
but did not return error code - thus we missed the fact that some
groups were not defined in breeze configuration.

(cherry picked from commit 01b69d3021536057a1b08347941f7211c2825d21)
